test/MUXControlsTestApp/App.xaml

```diff
@@ -35,6 +35,153 @@
                 <Setter Property="BorderThickness" Value="1"/>
                 <Setter Property="Padding" Value="8"/>
             </Style>
+
+            <Style TargetType="local:InfoBar" >
+                <Setter Property="Template">
+                    <Setter.Value>
+                        <ControlTemplate TargetType="local:InfoBar">
+                            <Border
+                                x:Name="Container"
+                                Height="70"
+                                Width="1000"
+                                VerticalAlignment="Top"
+                                Background="{TemplateBinding Background}"
+                                BorderBrush="Gray"
+                                BorderThickness="2">
+
+                                <VisualStateManager.VisualStateGroups>
+                                    <VisualStateGroup x:Name="ButtonStates">
+                                        <VisualState x:Name="NoButtonsVisible">
+                                            <VisualState.Setters>
+                                                <Setter Target="CloseButton.Visibility" Value="Collapsed"/>
+                                                <Setter Target="ActionButton.Visibility" Value="Collapsed"/>
+                                            </VisualState.Setters>
+                                        </VisualState>
+                                        <VisualState x:Name="CloseButtonVisible">
+                                            <VisualState.Setters>
+                                                <Setter Target="CloseButton.Visibility" Value="Visible"/>
+                                                <Setter Target="ActionButton.Visibility" Value="Collapsed"/>
+                                            </VisualState.Setters>
+                                        </VisualState>
+                                        <VisualState x:Name="ActionButtonVisible">
+                                            <VisualState.Setters>
+                                                <Setter Target="ActionButton.Visibility" Value="Visible"/>
+                                                <Setter Target="CloseButton.Visibility" Value="Collapsed"/>
+                                            </VisualState.Setters>
+                                        </VisualState>
+                                        <VisualState x:Name="BothButtonsVisible">
+                                            <VisualState.Setters>
+                                                <Setter Target="CloseButton.Visibility" Value="Visible"/>
+                                                <Setter Target="ActionButton.Visibility" Value="Visible"/>
+                                            </VisualState.Setters>
```

Comment:
I would recomend deleting the Setters for the default visual state. The way that Visual states work is to return any properties that changed to their defaults when you return to the default state so these lines are redundant.

Reply:
Ah, I see. I made the changes so now for each of my Visual State groups, there is a default visual state where there are no explicit setters.